bot/cogs/tags.py

```diff
@@ -28,21 +29,31 @@ class Tags(Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
         self.tag_cooldowns = {}
-
         self._cache = {}
-        self._last_fetch: float = 0.0
 
-    async def _get_tags(self, is_forced: bool = False) -> None:
+    @Cog.listener()
+    async def on_ready(self) -> None:
+        """Runs the code before the bot has connected."""
+        await self.get_tags()
+
+    async def get_tags(self) -> None:
         """Get all tags."""
-        # refresh only when there's a more than 5m gap from last call.
-        time_now: float = time.time()
-        if is_forced or not self._last_fetch or time_now - self._last_fetch > 5 * 60:
-            tags = await self.bot.api_client.get('bot/tags')
-            self._cache = {tag['title'].lower(): tag for tag in tags}
-            self._last_fetch = time_now
+        # Save all tags in memory.
+        tag_files = os.listdir("bot/resources/tags")
+        for file in tag_files:
+            p = Path("bot", "resources", "tags", file)
+            tag_title = os.path.splitext(file)[0].lower()
+            with p.open() as f:
+                tag = {
+                    "title": tag_title,
+                    "embed": {
+                        "description": f.read()
```

Comment:
As a precaution, this should add a gate for 2048 letter limits.
This is now a concern because we are no longer adding it via bot's command, which discord already limits to 2000 letters ( including the command's syntax )

Reply:
would it be best to paginate if it is over 2048?

Reply:
The problem lies in the `f.read()` - it can be a giant string greater than 2048, in theory, yes you can paginate it, and so liberate the limitation of length for each tag.

Reply:
We are no longer gonna paginate it as tags are meant to be a 1-page thing, more descriptive tags will be on the website.